commit 2290311d4d6d67618a06b0bf78aac0c96966fdda (refs/changes/44/1744/2)
Author: Dave Eddy <dave@daveeddy.com>
Date:   2017-04-07T12:23:06-04:00 (2 years, 6 months ago)
    
    OS-6033 node v0.10.26 error: stream.push() after EOF
    Reviewed by: Julien Gilli <julien.gilli@joyent.com>
    Approved by: Julien Gilli <julien.gilli@joyent.com>

diff --git a/node.js/Patches/node-f2b297c-stream-eof.diff b/node.js/Patches/node-f2b297c-stream-eof.diff
new file mode 100644
index 0000000..c7aed97
--- /dev/null
+++ b/node.js/Patches/node-f2b297c-stream-eof.diff
@@ -0,0 +1,14 @@
+diff --git a/lib/http.js b/lib/http.js
+index 3b1484b..0623668 100644
+--- a/lib/http.js
++++ b/lib/http.js
+@@ -344,9 +344,7 @@ IncomingMessage.prototype._read = function(n) {
+   // We actually do almost nothing here, because the parserOnBody
+   // function fills up our internal buffer directly.  However, we
+   // do need to unpause the underlying socket so that it flows.
+-  if (!this.socket.readable)
+-    this.push(null);
+-  else
++  if (this.socket.readable)
+     readStart(this.socket);
+ };
